src/Signal-Rohde&Schwarz_SMB100A/main.py

```diff
@@ -0,0 +1,329 @@
+# SOFTWARE.
+
+# SweepMe! driver
+# * Module: Signal
+# * Instrument: Rohde&Schwarz SMB100A
+
+import numpy as np
+from pysweepme.EmptyDeviceClass import EmptyDevice
+
+
+class Device(EmptyDevice):
+    """Device class for the Rohde&Schwarz SMB100A Signal Generator."""
+
+    def __init__(self) -> None:
+        """Initialize device parameters."""
+        EmptyDevice.__init__(self)
+
+        self.port_manager = True
+        self.port_types = [
+            "USB",
+            # "GPIB",
+            "TCPIP",
+        ]
+
+        self.variables = ["Frequency", "Amplitude"]
+        self.units = ["Hz", "V"]
+        self.plottype = [True, True]
+        self.savetype = [True, True]
+
+        # Measurement Parameters
+        self.sweep_mode: str = "None"
+        self.frequency_input: str = "Frequency in Hz"  # or "Period in s"
+        self.frequency: float = 2.2e9
+
+        self.amplitude_mode: str = "Amplitude in V"  # or "High level in V"
+        self.amplitude: float = 0.1
+
+        self.delay_mode: str = "Phase in deg"  # or "Delay in s"
+        self.delay: float = 0.0
+
+    def set_GUIparameter(self) -> dict:  # noqa: N802
+        """Set the standard GUI parameters."""
+        return {
+            "SweepMode": [
+                "None",
+                "Frequency in Hz",
+                "Period in s",
+                "Amplitude in V",
+                "High level in V",
+                "Phase in deg",
+                "Delay in s",
+            ],
+            "Waveform": ["Sinus"],
+            "PeriodFrequency": ["Frequency in Hz", "Period in s"],
+            "PeriodFrequencyValue": 2e-6,
+            "AmplitudeHiLevel": ["Amplitude in V", "High level in V"],
+            "AmplitudeHiLevelValue": 1.0,
+            "DelayPhase": ["Delay in s", "Phase in deg"],
+            "DelayPhaseValue": 0.0,
+            "Impedance": ["50"],
+        }
+
+    def get_GUIparameter(self, parameter: dict) -> None:  # noqa: N802
+        """Get the GUI parameters."""
+        self.sweep_mode = parameter["SweepMode"]
+        self.frequency_input = parameter["PeriodFrequency"]
+        self.frequency = float(parameter["PeriodFrequencyValue"])
+
+        self.amplitude_mode = parameter["AmplitudeHiLevel"]
+        self.amplitude = float(parameter["AmplitudeHiLevelValue"])
+
+        self.delay_mode = parameter["DelayPhase"]
+        self.delay = float(parameter["DelayPhaseValue"])
+
+        self.variables = [self.frequency_input.split(" ")[0], self.amplitude_mode.split(" ")[0]]
+        self.units = [self.frequency_input.split(" ")[-1], self.amplitude_mode.split(" ")[-1]]
+
+    def connect(self) -> None:
+        """Connect to the device."""
+        self.port.write("*RST")
+        # self.port.write("*CLS")
+
+    def configure(self) -> None:
+        """Configure the device."""
+        self.lock_user_interface(True)
+
+        self.set_power_mode("CW")
+        self.set_power_unit("V")
+
+        self.set_power_level(self.amplitude)
```

Comment:
This line suggest that we handover the amplitude, but it is still unclear whether the value is interpreted as amplitude or high level which is why it would make sense to give a variable name that reflects this. The function name set_power_level could be also changed to "handle_amplitude_highlevel" to make clear that is not known yet whether the power is set, especially because it also not a power that is set but rather a voltage.